### PR TITLE
Add API to get visible entries

### PR DESCRIPTION
**Pull request by MatthewKhouzam**
Allow only fetching data that is shown at the moment.

Signed-off-by: Matthew Khouzam <matthew.khouzam@ericsson.com>

### CODE REVIEW

**Comment by MatthewKhouzam** [comment]
Not yet ready, showing progress

File: timeline-chart/src/layer/time-graph-chart.ts [hunk 1]
```diff
@@ -214,8 +216,23 @@ export class TimeGraphChart extends TimeGraphChartLayer {
...
+
     protected updateScaleAndPosition() {
         if (this.rows) {
+            const canvas = this.canvas;
+            if (!canvas)
+                return;
+            const top = canvas.offsetTop;
+            const height = canvas.offsetHeight;
+            const bottom = top + height;
+            console.log("Bounds " + top + " - " + bottom);
```
Comment (PatrickTasse) [inline timeline-chart/src/layer/time-graph-chart.ts:234]: remove

File: timeline-chart/src/layer/time-graph-chart.ts [hunk 2]
```diff
@@ -228,6 +245,12 @@ export class TimeGraphChart extends TimeGraphChartLayer {
                         width: this.stateController.canvasDisplayWidth
                     }
                     comp.update(opts);
+                    const cTop = comp.position.y;
+                    const cBottom = comp.height + cTop;
+                    console.log("Object " + cTop + " - " + cBottom + " / " + top + " - " + bottom);
```
Comment (PatrickTasse) [inline timeline-chart/src/layer/time-graph-chart.ts:250]: remove

File: timeline-chart/src/layer/time-graph-chart.ts [hunk 3]
```diff
@@ -228,6 +245,12 @@ export class TimeGraphChart extends TimeGraphChartLayer {
...
+                    if (cTop <= bottom && top <= cBottom) {
```
Comment (PatrickTasse) [inline timeline-chart/src/layer/time-graph-chart.ts:251]: It appears that the vertical scrolling is not taken into account. The canvas bounds are unaffected by vertical scrolling, and the row y-position is fixed regardless of scrolling (e.g. it is the position in the full-height virtual scrollable area). Therefore it is always the first few rows of the model being considered visible, even when they are scrolled out of visible range.

**Review by PatrickTasse — CHANGES_REQUESTED** [review]

**Pull request closed without merging**